std/path.d

```diff
@@ -1850,6 +1755,30 @@ auto asNormalizedPath(R)(R path)
     return Result(path);
 }
 
+///
+unittest
+{
+    import std.array;
+    assert (asNormalizedPath("foo/..").array == ".");
+
+    version (Posix)
+    {
+        assert (asNormalizedPath("/foo/./bar/..//baz/").array == "/foo/baz");
+        assert (asNormalizedPath("../foo/.").array == "../foo");
+        assert (asNormalizedPath("/foo/bar/baz/").array == "/foo/bar/baz");
+        assert (asNormalizedPath("/foo/./bar/../../baz").array == "/baz");
+    }
+
+    version (Windows)
+    {
+        assert (asNormalizedPath(`c:\foo\.\bar/..\\baz\`).array == `c:\foo\baz`);
+        assert (asNormalizedPath(`..\foo\.`).array == `..\foo`);
+        assert (asNormalizedPath(`c:\foo\bar\baz\`).array == `c:\foo\bar\baz`);
+        assert (asNormalizedPath(`c:\foo\bar/..`).array == `c:\foo`);
+        assert (asNormalizedPath(`\\server\share\foo\..\bar`).array == `\\server\share\bar`);
```

Comment:
This line is too long; for ddoc'd unittest, we should try to keep line length <= 80 so that it's readable in the generated docs without needing to scroll sideways.